docs/docs/getting-started/developing-on-hydra.md

```diff
@@ -0,0 +1,89 @@
+# Hydra development
+
+This guide is meant to be a tutorial on how to develop an application on Hydra. It will assume the reader is already familiar with developing a DApp on Cardano and will focus on the differences between the two.
+
+## On-chain code
+
+On-chain code will be exactly the same between Cardano and a hydra head. This is one of the main selling points of the hydra design (heads are also called isomorphic state channels) the only caveat being that (at the time of writing 07-22) hydra heads do not support validity ranges for transactions, the reason for this is there is currently no notion of time inside of a hydra head.
+
+## Off-chain code
+
+Here is where the differences will begin. Usually, DApp developers will write some off-chain code to interact with their validators and minting policies.
+This off-chain code will usually be exposed through an HTTP interface that frontends can access to drive interactions with the application forward.
+Off-chain code is usually written in the Contract monad, which is used to build transactions and interact with the Cardano blockchain in a controlled way.
+
+When it comes to hydra, the developer will not have access to similar tools. The interface to speak to a hydra node is a websocket. Developers must build applications that connect to these websockets and can send and receive events through them.
+
+The Hydra node will emit several commands through this websocket: the API is defined [here](https://hydra.family/head-protocol/api-reference).
+The API reference is divided into two sections: Pub and Sub.
+The first one describes the commands that a hydra node will accept, while the second one describes the events that will be emitted through the websocket.
+
+An important event developers will be interested in is `SnaphotConfirmed`. This event is emitted once all the hydra nodes that are partaking in the head, have seen and signed a transaction submitted by one of them.
+
+Once a transaction lands inside a snapshot, it is confirmed and there are no chances for it to be rolled back. This is an important distinction from L1: in that case, when a node sees a transaction there is still a chance that it might be rolled back (where the probability of this goes to 0 as more blocks are added), in the hydra world, the only way to make a snapshot is to have every single node in the head approve of it.
```

Comment:
About rollbacks: this is only true after the head has been open long enough and the opening tx has become final. Might want to link this feature for more details: https://github.com/input-output-hk/hydra-poc/issues/185